pkg/apiserver/authz.go

```diff
@@ -63,21 +63,32 @@ const (
 // Keep this list in sync with constant list above.
 var AuthorizationModeChoices = []string{ModeAlwaysAllow, ModeAlwaysDeny, ModeABAC}
 
-// NewAuthorizerFromAuthorizationConfig returns the right sort of authorizer.Authorizer
-// based on the authorizationMode xor an error.  authorizationMode should be one of AuthorizationModeChoices.
-func NewAuthorizerFromAuthorizationConfig(authorizationMode string, authorizationPolicyFile string) (authorizer.Authorizer, error) {
-	if authorizationPolicyFile != "" && authorizationMode != "ABAC" {
-		return nil, errors.New("Cannot specify --authorization-policy-file without mode ABAC")
-	}
-	// Keep cases in sync with constant list above.
-	switch authorizationMode {
-	case ModeAlwaysAllow:
-		return NewAlwaysAllowAuthorizer(), nil
-	case ModeAlwaysDeny:
-		return NewAlwaysDenyAuthorizer(), nil
-	case ModeABAC:
-		return abac.NewFromFile(authorizationPolicyFile)
-	default:
-		return nil, errors.New("Unknown authorization mode")
+// NewAuthorizerFromAuthorizationConfig returns the right sort of union of multiple authorizer.Authorizer objects
+// based on the authorizationMode or an error.  authorizationMode should be a comma separated values
+// of AuthorizationModeChoices.
+func NewAuthorizerFromAuthorizationConfig(authorizationModes []string, authorizationPolicyFile string) (authorizer.Authorizer, error) {
```

Comment:
But I guess we should fail during startup itself rather than failing every request that comes to the authorizer after starting up the apiserver successfully.